packages/commerce-sdk-react/src/hooks/ShopperConsents/api.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useCallback} from 'react'
+import useAccessToken from '../useAccessToken'
+
+/**
+ * Custom ShopperConsents API client parameters
+ */
+export interface ShopperConsentsApiParams {
+    organizationId: string
+    siteId: string
+    locale: string
+}
+
+/**
+ * Subscription consent item
+ */
+export interface ConsentItem {
+    subscriptionId: string
+    contactPointValue: string
+    channel: string
+    status: string
+    title?: string
+    subtitle?: string
+    tags?: string[]
+}
+
+/**
+ * Custom ShopperConsents API client that uses the proxy endpoint
+ * TODO: Replace this with the official commerce-sdk-isomorphic client once available
+ */
+export const useShopperConsentsApiClient = (params: ShopperConsentsApiParams) => {
+    const {getTokenWhenReady} = useAccessToken()
+    const {organizationId, siteId, locale} = params
+
+    const apiCall = useCallback(
+        async (endpoint: string, method = 'GET', body: any = null) => {
+            const token = await getTokenWhenReady()
+            const baseUrl = `/mobify/proxy/api/shopper/shopper-consents/v1/organizations/${organizationId}`
+
+            const response = await fetch(`${baseUrl}${endpoint}`, {
+                method,
+                headers: {
+                    Authorization: `Bearer ${token}`,
+                    'Content-Type': 'application/json'
+                },
+                ...(body && {body: JSON.stringify(body)})
+            })
+
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`)
```

Comment:
```suggestion
                throw new Error(`HTTP error! status: ${response.status}`, {response}) 
```
For better traceability.